Nodejs/MigrateToJsps/MigrationLibrary.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Linq;
+using System.Security;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MigrateToJsps
+{
+    public class MigrationLibrary
+    {
+        public static void Migrate(string njsprojFile, string newProjectDir) 
+        {
+            if (string.IsNullOrEmpty(njsprojFile)) throw new ArgumentNullException("Please input a non-empty path to your .njsproj file.");
+
+            if (!string.IsNullOrEmpty(newProjectDir) && VerifyNewProjectDir(newProjectDir))
+            {
+                if (VerifyNjsprojPath(njsprojFile))
+                {
+                    MigrateProject(njsprojFile, newProjectDir);
+                }
+            }
+            else
+            {
+                throw new ArgumentException("Destination path not valid");
+            }
+        }
+
+        private static void MigrateProject(string njsprojFilepath, string destinationDir)
+        {
+            // TODO: move to JspsProjectCreator constructor
+            var njsprojFile = NjsprojFileReader.ProcessNjsproj(njsprojFilepath);
+
+            var njsprojDir = Path.GetDirectoryName(njsprojFilepath);
+            var renamedNjsprojDir = njsprojDir + "-old";
```

Comment:
Item #&#x2060;3 [above](https://github.com/microsoft/nodejstools/pull/2456#issue-1626392894) -- NTVS gets "-old", jsps gets the old name

Reply:
completed